app/workers/deletes.php

```diff
@@ -245,25 +249,56 @@ protected function deleteMemberships(Document $document, string $projectId): voi
 
     /**
      * @param Document $document project document
+     * @throws Exception
      */
     protected function deleteProject(Document $document): void
     {
         $projectId = $document->getId();
 
-        // Delete all DBs
-        $this->getProjectDB($projectId)->delete($projectId);
+        // Delete project tables
+        $dbForProject = $this->getProjectDB($projectId, $document);
+
+        $limit = 50;
+        $offset = 0;
+
+        while (true) {
+            $collections = $dbForProject->listCollections($limit, $offset);
+
+            if (empty($collections)) {
+                break;
+            }
+
+            foreach ($collections as $collection) {
+                $dbForProject->deleteCollection($collection->getId());
+            }
+
+            $offset += $limit;
```

Comment:
shouldn't after delete, the offset should still be 0?

Reply:
This is to delete all collections page by page, so 0-50, then 50-100 etc

Reply:
Oh now I see what you mean, you're right, will update